### PR TITLE
ARTEMIS-2320 Update javac-errorprone to 2.8.5 and error_prone_core to 2.3.3

### PR DESCRIPTION
**Pull request by jirkadanek**

### CODE REVIEW

**Comment by clebertsuconic** [comment]
lets wait the release before merging this please!

File: artemis-junit/src/test/java/org/apache/activemq/artemis/junit/ActiveMQConsumerResourceTest.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
    ActiveMQConsumerResource consumer = new ActiveMQConsumerResource(server.getVmURL(), TEST_QUEUE);
 
    @Rule
-   public RuleChain ruleChain = RuleChain.outerRule(new ThreadLeakCheckRule()).outerRule(server).around(consumer);
+   public RuleChain ruleChain = RuleChain.outerRule(new ThreadLeakCheckRule()).around(server).around(consumer);
```
Comment (jdanekrh) [inline artemis-junit/src/test/java/org/apache/activemq/artemis/junit/ActiveMQConsumerResourceTest.java]: Trying this on multiple machines, outside of debug mode it passes on my rhel vm, but not on bare metal laptop. When running under debugger, it passes everywhere. I have the same problem with other tests that use this rulechain construct, such as `MultipleEmbeddedActiveMQResourcesTest`.

```
*********************************************************************************
LEAKING THREADS
=============================================================================
Thread Thread[Attach Listener,9,system] is still alive with the following stackTrace:
*********************************************************************************
[main] 09:39:58,447 INFO  [org.apache.activemq.artemis.junit.ThreadLeakCheckRule] #test forceGC
[main] 09:39:58,500 INFO  [org.apache.activemq.artemis.junit.ThreadLeakCheckRule] #test forceGC Done 
*********************************************************************************
```

**Comment by jbertram** [comment]
This has been sitting for awhile, and now there is a conflict. @jdanekrh, can you resolve the conflict and we can get this merged once the 2.11.0 release is done.

**Comment by jdanekrh** [comment]
Conflict fixed. When this is merged, I'll do a 2.3.4 update. The release notes of error_prone promise better overall compile performance, which sounds attractive.

**Comment by jdanekrh** [comment]
(Tried that 2.3.4 and it is not any faster ;(

**Comment by jbertram** [comment]
This is defunct now. We're up to `error_prone_core` 2.6.0 now. All the tests have been fixed by other commits as well.

**Pull request closed without merging**